### PR TITLE
add loading multiple datasets

### PR DESCRIPTION
**Pull request by AgaMiko**
Add option to load multiple datasets.
To load multiple datasets simply pass to `args.parser` list of datasets (file paths to videos separated with spaces)

`python3 train.py --data "/dih4/dih4_2/hearai/data/frames/pjm" "/dih4/dih4_2/hearai/data/frames/basic_lexicon" --epochs 100 --lr 1e-4 --classification-mode "hamnosys" --neptune --num_segments 16 --b 4 --workers 0 --gpu 1`

Attention! The method requires that all `annotation_file`s have exactly the same annotation filename e.g. `"test_hamnosys.txt"`! If you need to pass different path you need to do it manually.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -18,7 +18,8 @@ def get_args_parser():
     parser = argparse.ArgumentParser()
     # Data parameters and paths
     parser.add_argument(
-        "--data", help="path to data", default="assets/sanity_check_data",
+        "--data", help="path to data", nargs="*",
+        default=["assets/sanity_check_data"],
```
Comment (AgaMiko) [inline train.py:22]: add `nargs="*"` to load multiple arguments

File: train.py [hunk 2]
```diff
@@ -107,14 +108,8 @@ def main(args):
 
     # set torch seed
     torch.manual_seed(args.seed)
-
-    # load data
-    videos_root = args.data
```
Comment (AgaMiko) [inline train.py:112]: moved it down

File: train.py [hunk 3]
```diff
@@ -124,17 +119,31 @@ def main(args):
...
+            classification_mode=args.classification_mode,
+            num_segments=args.num_segments,
+            time=args.time,
+            landmarks_path=args.landmarks_path,
+            transform=preprocess,
+            test_mode=True,
+        )
+        datasets.append(dataset)
+    dataset = torch.utils.data.ConcatDataset(datasets)
+
```
Comment (AgaMiko) [inline train.py:146]: load all datasets and concat them

**Review by majsylw — APPROVED** [review]

File: datasets/dataset.py [hunk 4]
```diff
@@ -20,6 +20,14 @@
     reload(sys)
     sys.setdefaultencoding("utf-8")
 
+ALL_HAMNOSYS_HEADS = {"symmetry_operator": 3,
```
Comment (AgaMiko) [inline datasets/dataset.py:23]: I wouldn't hardcode it, I would suggest simply `create_heads_dict(classification_mode)` but with "hamnosys" mode if you need a full hamnosys dict. Otherwise, any change in a dict requires unnecessary changes in two places in the code.

File: utils/classification_mode.py [hunk 5]
```diff
@@ -6,14 +6,19 @@ def create_heads_dict(classification_mode):
...
+            "hand_shape_base_form": 12,
             "hand_shape_thumb_position": 4,
             "hand_shape_bending": 6,
             "hand_position_finger_direction": 18,
             "hand_position_palm_orientation": 8,
             "hand_location_x": 5,
-            "hand_location_y": 36,
+            "hand_location_y": 37,
+        }  # number of classes for each head
+    elif classification_mode == "hamnosys-less":
```
Comment (AgaMiko) [inline utils/classification_mode.py:18]: maybe classification_mode "custom" instead, and loading `dict` from a `json` form selected path? It would be easier, and wouldn't require adding new heads or modifying  the code each time for every tested combination. What do you think?

Comment (mariaogryczak) [reply utils/classification_mode.py:18]: I agree with this idea. It's still changing and it would be easier to use it this way while the work is still in progress.

Comment (majsylw) [reply utils/classification_mode.py:18]: it is very good idea, but maybe it is better to introduce it here #48 after merging this PR?

File: datasets/dataset.py (hunk 4 above)
Comment (majsylw) [reply datasets/dataset.py:23]: agree, however this dict is different it maps name of head with number for collumn, we can assume that it is always the same and map this somewhere else, what do you think @natcz?

Comment (AgaMiko) [reply datasets/dataset.py:23]: Then I suggest adding headers to annotation files. If the headers are exactly the same as in the dict, then it is much easier to locate each column, without the need of writing it in code. E.g. load header `hand_shape_bending` and dynamically make such a dictionary

File: utils/classification_mode.py (hunk 5 above)
Comment (AgaMiko) [reply utils/classification_mode.py:18]: I would add it here as this is a minimal change in code. Requires only passing a path to a json file, and loading it
```
with open(path) as json_file:
    num_classes_dict = json.load(json_file)
```

File: datasets/dataset.py (hunk 4 above)
Comment (majsylw) [reply datasets/dataset.py:23]: it can be harder to introduce without major changes in the code to use this type of header, _parse_annotation function has to be changed to check if header exists and after this step header can be read from file

**Review by mariaogryczak — APPROVED** [review]

File: datasets/dataset.py (hunk 4 above)
Comment (AgaMiko) [reply datasets/dataset.py:23]: ok, so next sprint maybe?
